docs/pulses.ipynb

```diff
@@ -0,0 +1,251 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "6b645603-12c0-4b3d-ab3c-529ad4f373e3",
+   "metadata": {},
+   "source": [
+    "# Creating pulses\n",
+    "\n",
+    "This notebook will illustrate how to create different kinds of pulses for the beamline."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "899a245a-0ae7-45dd-b684-3b9d1c034cc6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import scipp as sc\n",
+    "import tof\n",
+    "\n",
+    "AA = sc.Unit('angstrom')\n",
+    "ms = sc.Unit('ms')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3890d24c-93ab-4faa-b653-c9208b5eba23",
+   "metadata": {},
+   "source": [
+    "## A basic pulse\n",
+    "\n",
+    "Pulses are characterized by three aspects:\n",
+    "\n",
+    "- the number of neutrons in the pulse\n",
+    "- the time at which each neutron is born inside the pulse\n",
+    "- the wavelength of each neutron inside the pulse\n",
+    "\n",
+    "By default, the time and wavelength distributions of neutrons are both flat (uniform).\n",
+    "We then need to define a pulse duration (start and end times) and a wavelength range.\n",
+    "\n",
+    "To create a pulse with 1M neutrons, uniformly distributed in the ranges of 1-3 ms for birth times,\n",
+    "and 1-10 Å for wavelengths, we write:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a2699065-309c-439c-bcda-bf15d20edd0b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pulse = tof.Pulse(\n",
+    "    neutrons=1_000_000, tmin=1.0 * ms, tmax=3.0 * ms, lmin=1.0 * AA, lmax=10.0 * AA\n",
```

Comment:
What is `l`?

Reply:
lambda. I'm happy to change. Initially I felt `wavelength_min/wavelength_max` was a bit lengthy, but I am probably changing my mind in the favour of clarity.

Reply:
`wmin`, `wmax` might be marginally clearer? But I agree that being lengthy is probably better.